Add JS & CSS lint actions on PRs

RahiDroid opened the pull request:
## Description
Adds JS & CSS lint checks on PRs
<!-- Very brief idea on what this PR does? No technical details. -->

## Technical Details

<!-- Any technical details that should be explained or mentioned to the reviewers. -->

## Checklist

<!--
If the ticket you worked on had any checklist, include that here and mark items that are covered in this PR. Else, create a checklist of all the items this PR covers.

Example:
- [ ] Fix incorrect meta key value for the Contact block.
- [ ] Fix footer button alignment issue.
-->

## Screenshots

<!-- Add screenshots or screen recordings if applicable and required. -->

## To-do

<!-- Mention anything that needs to be done after this PR as a follow-up or if anything is left uncovered. -->

## Fixes/Covers issue

<!-- Add the issue number which the PR is intended to be for. -->
Fixes #

<!-- After you're done creating the PR, assign the PR to yourself and assign reviewers for the PR. If unsure about whom to ask for a review, leave blank and ping on the project-specific channel. -->

thelovekesh commented on .github/workflows/lint-js-css.yml:
```diff
@@ -0,0 +1,43 @@
+
+      - name: CSS Lint
+        run: npm run lint:css
+
+      - name: JS Lint
+        run: npm run lint:js
```
```suggestion
```

thelovekesh replied:
This is a duplicate entry.

thelovekesh commented on .github/workflows/lint-js-css.yml:
```diff
@@ -0,0 +1,43 @@
+
+on:
+  pull_request:
+    paths:
```
Do we need to require paths explicitly? We already have specified paths in the `npm:lint*` commands.

thelovekesh reviewed (commented):
I have a few more ideas in mind which can be implemented in this workflow:
- We can run linting for the package.json using `npm run lint:pkg-json`, so it's makes sense to add it as well.
- `actions/cache@v3` GHA can be used for node_modules cache.
- Add concurrency so that previous tests can be canceled if new changes pushed recently.
- We can specify when to run a workflow on PR i.e.
```yml
  pull_request:
    types:
      - synchronize
      - opened
      - ready_for_review
```
Should we run this workflow on Node 14 as well?

thelovekesh commented:
@RahiDroid Should we include JS tests in this workflow as well?

RahiDroid commented:
> We can run linting for the package.json using npm run lint:pkg-json, so it's makes sense to add it as well.
Added!

> actions/cache@v3 GHA can be used for node_modules cache.
Actions documentation states that, "It is not recommended to cache node_modules, as it can break across Node versions and won’t work with npm ci".

Add concurrency so that previous tests can be canceled if new changes pushed recently.
> Added!

> We can specify when to run a workflow on PR 
Yup, adding `ready_for_review` makes sense, rest are default. Added!

> Should we run this workflow on Node 14 as well
I don't think it's necessary.

> @RahiDroid Should we include JS tests in this workflow as well?
Should be part of different workflow but will add in this PR
